frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/LogsTab.tsx

```diff
@@ -0,0 +1,245 @@
+    <Button
+      variant={!loaded ? 'plain' : isPaused ? 'plain' : 'link'}
+      onClick={() => {
+        setIsPaused(!isPaused);
+      }}
+    >
+      {!loaded ? <Spinner size="sm" /> : isPaused ? <PlayIcon /> : <PauseIcon />}
+      {!loaded ? ` Loading log` : isPaused ? ` Resume refreshing` : ` Pause refreshing`}
+    </Button>
+  );
+
+  const leftAlignedToolbarGroup = (
+    <React.Fragment>
+      <ToolbarToggleGroup toggleIcon={<EllipsisVIcon />} breakpoint="md">
+        <ToolbarItem>
+          <TextContent>
+            <TextListItem component={TextListItemVariants.dt}>Container</TextListItem>
+          </TextContent>
+        </ToolbarItem>
+        <ToolbarItem variant="search-filter">
+          <SimpleDropdownSelect
+            options={podContainers.map((container) => ({
+              key: container.name,
+              label: container.name,
+            }))}
+            value={selectedContainer?.name ?? ''}
+            showTooltipValue={true}
+            placeholder="Select container..."
+            onChange={(v) => setSelectedContainer(podContainers.find((c) => c.name === v) ?? null)}
+          />
+        </ToolbarItem>
+      </ToolbarToggleGroup>
+      <ToolbarItem>
+        <ControlButton />
+      </ToolbarItem>
+    </React.Fragment>
+  );
+
+  const rightAlignedToolbarGroup = (
+    <React.Fragment>
+      <ToolbarGroup variant="icon-button-group">
+        <ToolbarItem>
+          <Tooltip position="top" content={<div>Download</div>}>
+            <Button onClick={onDownloadClick} variant="link" aria-label="Download current logs">
+              <DownloadIcon />
+              Download
+            </Button>
+          </Tooltip>
+        </ToolbarItem>
+      </ToolbarGroup>
+    </React.Fragment>
+  );
+
+  const FooterButton = () => {
+    const handleClick = () => {
+      setIsPaused(false);
+    };
+    return (
+      <Button onClick={handleClick} isBlock>
+        <OutlinedPlayCircleIcon />
+        Resume refreshing log
+      </Button>
+    );
+  };
+  return (
+    <>
+      {!loaded ? (
+        <Spinner />
+      ) : error ? (
+        <Alert
+          isExpandable
+          isInline
+          variant="danger"
+          title="An error occurred while retrieving the requested logs"
+        >
+          <p>{error.message}</p>
+          <p>
+            <a onClick={refresh}>Retry</a>
+          </p>
+        </Alert>
+      ) : (
+        <Alert
+          isExpandable
+          isInline
+          variant="warning"
+          title="The log window displays partial content"
+        >
+          <p>
+            The log refreshes every 3 seconds and displays the latest 500 lines. Exceptionally long
+            lines are abridged. To view the full log for this task, you can{' '}
+            <a onClick={onDownloadClick}>download all container logs</a> associated with it.
+          </p>
+        </Alert>
+      )}
+      <LogViewer
+        key={reRenderKey}
```

Comment:
I initially added this because the logs were only rendering when the window size was changed. So I had to add this. I was debugging this with @DaoDaoNoCode

Reply:
Ah yes, we were trying to debug and figure out why it could not render normally. Maybe try `mountOnEnter` instead.

Reply:
Wait -- what? The logs were only rendering when the window size changed? Why didn't you ping me? Don't hack things, ask for help. The team has a lot of knowledge on React.

Reply:
No, it's not very exact. The problem should be:
1. We render the logs tab and hide it too early and the log viewer cannot successfully calculate the position of every single line, which causes the render problem. When you change the window size (the browser window size), it will trigger the built-in listener on the log viewer, which will cause a re-calculation of the log positions, and can render everything successfully. However, if we mount the log viewer when entering the logs tab, this could also render successfully because the calculation could be done correctly at that time.
2. The listener of re-calculation is a built-in one as I mentioned above, so the log height and position will not be re-calculated when you change the drawer side panel size (not the browser window size), so if you expand or shrink the drawer, the line height of the logs could be problematic. If we can have a listener to listen to the drawer side panel size change and re-render the log viewer, the problem could be solved. (It may also possibly solve the problem in 1^, not sure, needs to verify)

Reply:
Tried `mountOnEnter`. Didn't work for me.

Reply:
Can you start a thread in Slack @manaswinidas, this back and forth isn't going to be speedy if we do it here. Let us talk about what your scenario is and look at the problem fresh.